bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GCData.java

```diff
@@ -57,4 +60,13 @@ public final class GCData {
 	public float gdipXOffset, gdipYOffset;
 	public int uiState = 0;
 	public boolean focusDrawn;
+
+	int getNativeDeviceZoom() {
+		return nativeDeviceZoom;
+	}
+
+	public void setNativeDeviceZoom(int nativeDeviceZoom) {
+		this.nativeDeviceZoom = nativeDeviceZoom;
+		this.deviceZoom = DPIUtil.getZoomForAutoscaleProperty(nativeDeviceZoom);
+	}
```

Comment:
As discussed, adding a public method to a platform-dependent, public class is only allowed when added to all three implementations. Or do I miss anything why that's not necessary here, @akoch-yatta @amartya4256?

Reply:
The whole class of GCData is only applicable to windows. So I think it should be okay to have a public method inside. I and @fedejeanne had a talk about this change before. Let me know what you think.

Reply:
Here it reads as follows:
`/**
 * Instances of this class are descriptions of GCs in terms
 * of unallocated platform-specific data fields.
 * <p>
 * <b>IMPORTANT:</b> This class is <em>not</em> part of the public
 * API for SWT. It is marked public only so that it can be shared
 * within the packages provided by SWT. It is not available on all
 * platforms, and should never be called from application code.
 * </p>
 * @see <a href="http://www.eclipse.org/swt/">Sample code and further information</a>
 * @noreference This class is not intended to be referenced by clients
 */
public final class GCData {`

Reply:
I think @HeikoKlare is correct about this remark. Although the GCData is heavily different on all OS, the difference is only on attribute level. We discussed about adding the method to simplify the usage (by setting two attributes via one call), but we mustn't do that as a publich method.

Reply:
If I remember right, API checks are not enabled for SWT in Jenkins, but in the IDE. @HannesWell : please correct me if I'm wrong. So the correct way to review SWT PR that might change API is to import PR in the IDE and clean build SWT there.

Reply:
If I am not mistaken, API checks in CI / Jenkins have been recently enabled: https://github.com/eclipse-platform/eclipse.platform.swt/pull/1011 
But these checks only validate versions and not the "compatibility" between the different OS implementations, do they? So if the minor of SWT has already been bumped in the current development cycle and if you then add public API only for a single OS implementation, I would not expect any automated check to complain. This error will only be caught "automatically" when a consumer of that public API method is added in code that is compiled for all operating systems. Then this will result in a compiler error because of the according method not existing in the class for that specific OS.

Reply:
So if I read this correctly, the only way to see if we're breaking the API these days is to:
1. Use the newly added code (or all the public API for that matter) in client code, _e.g._ in tests inside platform-independent plugins like `org.eclipse.swt.tests`
2. Compile for all platforms --> Jenkins

(+1 for TDD and full test coverage )

And also: the I-Builds would **not** break if one forgets to add the methods in some platforms.

right, @HeikoKlare ?

@iloveeclipse I didn't quite get what you meant by compiling it inside the IDE: does this help to catch inconsistencies in the public API across platforms or merely breaking API changes in the current platform?

Reply:
> I didn't quite get what you meant by compiling it inside the IDE: does this help to catch inconsistencies in the public API across platforms or merely breaking API changes in the current platform?

The later, sorry for confusion. The consistency between platform API's is supposed to be reviewed manually by the reviewer.

> If I am not mistaken, API checks in CI / Jenkins have been recently enabled: https://github.com/eclipse-platform/eclipse.platform.swt/pull/1011

Good. I'm not sure it was only done for Linux or for every platform.